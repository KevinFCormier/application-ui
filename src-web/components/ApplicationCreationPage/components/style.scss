/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2020. All Rights Reserved.
 * Copyright (c) 2020 Red Hat, Inc.
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
@import 'node_modules/carbon-components/scss/globals/scss/_colors';
@import 'scss/variables';

.timeWindow-container {
  position: relative;

  .timeWindow-btn-group {
    display: inline-grid;
  }

  .timeWindow-btn {
    margin: 5px 0;
  }

  .timeWindow-config-days-checkboxes {
    width: 50%;
    display: grid;
    grid-template-columns: auto auto;

    .first-col {
      grid-column-start: 1;
    }

    .second-col {
      grid-column-start: 2;
    }
  }

  .timeWindow-config-timezone-dropdown {
    width: 180px;
  }

  .timeWindow-config-start-end-times {
    display: block;
  }

  .timeWindow-config-start {
    float: left;
  }

  .bx--select--inline {
    svg.bx--select__arrow {
      position: relative;
    }
  }
}