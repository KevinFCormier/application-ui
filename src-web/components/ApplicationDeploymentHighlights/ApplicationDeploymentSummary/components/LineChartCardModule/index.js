/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/

import React from 'react'
import { withLocale } from '../../../../../providers/LocaleProvider'
import {
  BarChart,
  Bar,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
  ResponsiveContainer
} from 'recharts'
import {
  getChartKeyName,
  getModuleData,
  getMaxStringWidth,
  toPercent
} from './utils'

const LineChartCardModule = withLocale(({ data, locale }) => {
  const moduleData = getModuleData(data)
  const maxString = getMaxStringWidth(data) * 2 + 30
  const domain = [0, 1]
  const ticks = [0, 0.2, 0.4, 0.6, 0.8, 1]

  const renderTooltipContent = line => {
    const { payload } = line
    if (payload && payload.length > 0) {
      const entry = payload[0]
      return (
        <div className="customized-tooltip-content">
          <span className="bx--tooltip__caret" />
          <p className="total">{`${entry.payload.tooltip_name}`}</p>
          <ul className="list">
            {
              <li key={'item-0'} style={{ color: entry.color }}>
                {`${entry.name}: ${entry.payload.completed} / ${
                  entry.payload.total
                }`}
              </li>
            }
          </ul>
        </div>
      )
    }
  }

  return (
    <ResponsiveContainer width="95%" height="90%">
      <BarChart
        layout="vertical"
        width={400}
        height={250}
        data={moduleData.chartCardItems}
        margin={{
          top: 40,
          right: 20,
          left: maxString,
          bottom: 20
        }}
      >
        <CartesianGrid strokeDasharray="0" vertical={false} stroke="#F5F7FA" />
        <defs>
          <linearGradient id="colorUv" x1="1" y1="0" x2="0" y2="0">
            <stop offset="0%" stopColor="#00BAB6" stopOpacity={0.9} />
            <stop offset="100%" stopColor="#47ECEB" stopOpacity={0.6} />
          </linearGradient>
        </defs>
        <defs>
          <linearGradient id="colorNotCompl" x1="1" y1="0" x2="0" y2="0">
            <stop offset="5%" stopColor="#F5F7FA" stopOpacity={0.9} />
            <stop offset="95%" stopColor="#F5F7FA" stopOpacity={0.6} />
          </linearGradient>
        </defs>
        <XAxis
          type="number"
          axisLine={false}
          tickFormatter={toPercent}
          tickCount={6}
          domain={domain}
          ticks={ticks}
          tick={{ fontSize: 10 }}
          tickLine={false}
        />
        <YAxis
          type="category"
          dataKey="name"
          tick={{ fontSize: 10 }}
          interval="preserveStartEnd"
          axisLine={false}
          tickLine={false}
        />
        <Tooltip content={renderTooltipContent} />
        <Bar
          barSize={8}
          legendType="circle"
          dataKey="percent_completed"
          stackId="a"
          fillOpacity={1}
          fill="url(#colorUv)"
          name={getChartKeyName('percent_completed', locale)}
        />
        <Bar
          barSize={8}
          legendType="circle"
          dataKey="percent_not_completed"
          stackId="a"
          fillOpacity={1}
          fill="url(#colorNotCompl)"
          name={getChartKeyName('percent_not_completed', locale)}
        />
      </BarChart>
    </ResponsiveContainer>
  )
})

LineChartCardModule.propTypes = {}

export default withLocale(LineChartCardModule)
