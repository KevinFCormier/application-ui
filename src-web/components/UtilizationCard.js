/*******************************************************************************
 * Licensed Materials - Property of IBM
 * 5737-E67
 * (c) Copyright IBM Corporation 2018. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/

import React from 'react'
import PropTypes from 'prop-types'
import { Module, ModuleHeader, ModuleBody } from 'carbon-addons-cloud-react'
import resources from '../../lib/shared/resources'
import msgs from '../../nls/platform.properties'
import { Link } from 'react-router-dom'
import config from '../../lib/shared/config'

resources(() => {
  require('../../scss/dashboard-card.scss')
})

const DashboardOrb = ({
  status = 'healthy',
  value,
  utilStatus,
  className = ''
}) => (
  <div className={`dashboard-count ${className}`}>
    <div className="dashboard-orb">
      {value === 0 ? (
        <div
          className={'dashboard-orb dashboard-orb__inner dashboard-orb__gray'}
        >
          {value}
        </div>
      ) : (
        <div
          className={`dashboard-orb dashboard-orb__inner dashboard-orb__util-${status}`}
        >
          {value}
        </div>
      )}
    </div>
    {value === 0 ? (
      <p className={'dashboard-card-text dashboard-card-text__gray'}>
        {utilStatus}
      </p>
    ) : (
      <p className={'dashboard-card-text'}>{utilStatus}</p>
    )}
  </div>
)

const OrbPropType = {
  className: PropTypes.string,
  status: PropTypes.oneOf(['critical', 'warning', 'healthy']),
  utilStatus: PropTypes.string,
  value: PropTypes.number
}

DashboardOrb.propTypes = OrbPropType

export class UtilizationCard extends React.PureComponent {
  render() {
    const { locale } = this.context
    const { critical = 0, healthy = 0, title, warning = 0 } = this.props
    return (
      <Module
        className={'bx--tile dashboard-card'}
        size="single"
        {...this.props}
      >
        <ModuleHeader className={'dashboard-card-header__utilization'}>
          {title}
        </ModuleHeader>
        <ModuleBody className={'dashboard-card-orb'}>
          <div className="dashboard-overview">
            <div>
              <DashboardOrb
                key="critical-orb"
                status="critical"
                utilStatus={msgs.get(
                  'dashboard.utilization.orb.critical',
                  locale
                )}
                value={critical}
              />
              <DashboardOrb
                key="warning-orb"
                status="warning"
                utilStatus={msgs.get(
                  'dashboard.utilization.orb.warning',
                  locale
                )}
                value={warning}
              />
              <DashboardOrb
                key="healthy-orb"
                status="healthy"
                utilStatus={msgs.get(
                  'dashboard.utilization.orb.healthy',
                  locale
                )}
                value={healthy}
              />
            </div>
          </div>
        </ModuleBody>
        <ModuleBody className={'dashboard-card-table-body'}>
          <div className={'dashboard-card-separator-link'}>
            <Link to={`${config.contextPath}/clusters`}>
              {msgs.get('dashboard.module.separator.link')}
            </Link>
          </div>
        </ModuleBody>
      </Module>
    )
  }
}

UtilizationCard.propTypes = {
  critical: PropTypes.number,
  healthy: PropTypes.number,
  title: PropTypes.string,
  warning: PropTypes.number
}
