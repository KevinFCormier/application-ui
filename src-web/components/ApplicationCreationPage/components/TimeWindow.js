/*******************************************************************************
 * Licensed Materials - Property of IBM
 * Copyright (c) 2020 Red Hat, Inc. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import PropTypes from 'prop-types'
import resources from '../../../../lib/shared/resources'
import {
  Accordion,
  AccordionItem,
  Checkbox,
  DropdownV2,
  Icon,
  RadioButton,
  RadioButtonGroup,
  SelectItem,
  TimePicker,
  TimePickerSelect
} from 'carbon-components-react'
import Tooltip from '../../TemplateEditor/components/Tooltip'
import msgs from '../../../../nls/platform.properties'

resources(() => {
  require('./style.scss')
})

class TimeWindow extends React.Component {
  static propTypes = {
    control: PropTypes.object,
    handleChange: PropTypes.func,
    locale: PropTypes.string
  };

  constructor(props) {
    super(props)
    this.state = {}
    this.props.control.active = { mode: '', days: [], timezone: '' }
  }

  render() {
    const { locale, control } = this.props
    const { name, validation = {} } = control
    const modeSelected = control.active.mode ? true : false
    const daysSelectorID = 'days-selector'
    const timezoneDropdownID = 'timezone-dropdown'

    return (
      <React.Fragment>
        <div className="creation-view-controls-labels">
          <div className="creation-view-controls-textarea-title">
            {name}
            {validation.required ? (
              <div className="creation-view-controls-required">*</div>
            ) : null}
            <Tooltip control={control} locale={locale} />
          </div>

          <div className="timeWindow-container">
            <RadioButtonGroup
              className="timeWindow-mode-container"
              name="timeWindow-mode-container"
            >
              <RadioButton
                className="mode-btn"
                id="active-mode"
                labelText={msgs.get(
                  'creation.app.settings.timeWindow.activeMode',
                  locale
                )}
                value="&quot;active&quot;"
                onClick={this.handleChange.bind(this)}
              />
              <RadioButton
                className="mode-btn"
                id="block-mode"
                labelText={msgs.get(
                  'creation.app.settings.timeWindow.blockMode',
                  locale
                )}
                value="&quot;block&quot;"
                onClick={this.handleChange.bind(this)}
              />
            </RadioButtonGroup>

            <Accordion>
              <AccordionItem
                open={modeSelected}
                title={msgs.get(
                  'creation.app.settings.timeWindow.config',
                  locale
                )}
              >
                <div className="timeWindow-config-container">
                  <div className="config-days-section">
                    <div className="config-title">
                      {msgs.get(
                        'creation.app.settings.timeWindow.config.days.title',
                        locale
                      )}{' '}
                      *
                    </div>
                    <div className="config-descr">
                      {msgs.get(
                        'creation.app.settings.timeWindow.config.days.descr',
                        locale
                      )}
                    </div>
                    <div className="config-days-selector">
                      <div className="first-col">
                        <Checkbox
                          labelText="Monday"
                          name={daysSelectorID}
                          id="mon"
                          value="&quot;Monday&quot;"
                          disabled={!modeSelected}
                          onClick={this.handleChange.bind(this)}
                        />
                        <Checkbox
                          labelText="Tuesday"
                          name={daysSelectorID}
                          id="tues"
                          value="&quot;Tuesday&quot;"
                          disabled={!modeSelected}
                          onClick={this.handleChange.bind(this)}
                        />
                        <Checkbox
                          labelText="Wednesday"
                          name={daysSelectorID}
                          id="wed"
                          value="&quot;Wednesday&quot;"
                          disabled={!modeSelected}
                          onClick={this.handleChange.bind(this)}
                        />
                        <Checkbox
                          labelText="Thursday"
                          name={daysSelectorID}
                          id="thurs"
                          value="&quot;Thursday&quot;"
                          disabled={!modeSelected}
                          onClick={this.handleChange.bind(this)}
                        />
                        <Checkbox
                          labelText="Friday"
                          name={daysSelectorID}
                          id="fri"
                          value="&quot;Friday&quot;"
                          disabled={!modeSelected}
                          onClick={this.handleChange.bind(this)}
                        />
                      </div>
                      <div className="second-col">
                        <Checkbox
                          labelText="Saturday"
                          name={daysSelectorID}
                          id="sat"
                          value="&quot;Saturday&quot;"
                          disabled={!modeSelected}
                          onClick={this.handleChange.bind(this)}
                        />
                        <Checkbox
                          labelText="Sunday"
                          name={daysSelectorID}
                          id="sun"
                          value="&quot;Sunday&quot;"
                          disabled={!modeSelected}
                          onClick={this.handleChange.bind(this)}
                        />
                      </div>
                    </div>
                  </div>

                  <div className="config-timezone-section">
                    <div className="config-title">
                      {msgs.get(
                        'creation.app.settings.timeWindow.config.timezone.title',
                        locale
                      )}{' '}
                      *
                    </div>
                    <DropdownV2
                      className="config-timezone-dropdown"
                      label="Choose a location"
                      items={[
                        {
                          label: 'America/Edmonton',
                          name: timezoneDropdownID,
                          value: '"America/Edmonton"'
                        },
                        {
                          label: 'America/Halifax',
                          name: timezoneDropdownID,
                          value: '"America/Halifax"'
                        },
                        {
                          label: 'America/Toronto',
                          name: timezoneDropdownID,
                          value: '"America/Toronto"'
                        },
                        {
                          label: 'America/Vancouver',
                          name: timezoneDropdownID,
                          value: '"America/Vancouver"'
                        },
                        {
                          label: 'America/Winnipeg',
                          name: timezoneDropdownID,
                          value: '"America/Winnipeg"'
                        }
                      ]}
                      disabled={!modeSelected}
                      onChange={this.handleChange.bind(this)}
                    />
                  </div>

                  <div className="config-time-section">
                    <div className="config-start-time">
                      <TimePicker
                        id="start-time"
                        labelText="Start Time"
                        maxLength={5}
                        placeholder="hh:mm"
                        type="text"
                        disabled={!modeSelected}
                      >
                        <TimePickerSelect
                          id="start-time-selector"
                          labelText="AM/PM"
                          placeholder="hh:mm"
                          disabled={!modeSelected}
                        >
                          <SelectItem
                            disabled={false}
                            hidden={false}
                            text="AM"
                            value="AM"
                          />
                          <SelectItem
                            disabled={false}
                            hidden={false}
                            text="PM"
                            value="PM"
                          />
                        </TimePickerSelect>
                      </TimePicker>
                    </div>
                    <div className="config-end-time">
                      <TimePicker
                        id="end-time"
                        labelText="End Time"
                        maxLength={5}
                        placeholder="hh:mm"
                        type="text"
                        disabled={!modeSelected}
                      >
                        <TimePickerSelect
                          id="end-time-selector"
                          labelText="AM/PM"
                          placeholder="hh:mm"
                          disabled={!modeSelected}
                        >
                          <SelectItem
                            disabled={false}
                            hidden={false}
                            text="AM"
                            value="AM"
                          />
                          <SelectItem
                            disabled={false}
                            hidden={false}
                            text="PM"
                            value="PM"
                          />
                        </TimePickerSelect>
                      </TimePicker>
                    </div>
                    <div className="add-time-btn">
                      <Icon
                        name="icon--add--glyph"
                        fill="#3d70b2"
                        description=""
                        className="add-time-btn-icon"
                      />
                      <div className="add-time-btn-text">
                        {msgs.get(
                          'creation.app.settings.timeWindow.config.time.add',
                          locale
                        )}
                      </div>
                    </div>
                  </div>
                </div>
              </AccordionItem>
            </Accordion>

            {/* <TextArea
              id={id}
              name="timeWindow-text"
              invalid={!!exception}
              invalidText={exception}
              hideLabel
              spellCheck={false}
              autoComplete={'no'}
              labelText=""
              onChange={this.handleChange.bind(this)}
            /> */}
          </div>
        </div>
      </React.Fragment>
    )
  }

  handleChange(event) {
    const { control, handleChange } = this.props
    let targetName = ''
    try {
      targetName = event.target.name
    } catch (e) {
      targetName = ''
    }

    if (targetName && targetName === 'timeWindow-mode-container') {
      control.active.mode = event.target.value
    } else if (targetName && targetName === 'days-selector') {
      if (event.target.checked === true) {
        control.active.days.push(event.target.value)
      } else {
        const index = control.active.days.indexOf(event.target.value)
        control.active.days.splice(index, 1)
      }
    } else if (
      event.selectedItem &&
      event.selectedItem.name === 'timezone-dropdown'
    ) {
      control.active.timezone = event.selectedItem.value
    }

    // else if (event.target.name === 'timeWindow-text') {
    //   control.active.text = '"' + event.target.value + '"'
    // }
    handleChange(control)
  }
}

export default TimeWindow
