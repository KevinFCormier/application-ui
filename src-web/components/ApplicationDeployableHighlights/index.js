/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/

import React from '../../../node_modules/react';
import CountsCardModule from '../CountsCardModule';
import msgs from '../../../nls/platform.properties';
import { connect } from '../../../node_modules/react-redux';
import resources from '../../../lib/shared/resources';

resources(() => {
  require('./style.scss');
});

const mapStateToProps = (state) => {
  const { } = state;

  const countsCardDataGeneralInfo = [
    {
      msgKey: 'dashboard.card.deployable.versions',
      count: 5,
    },
    {
      msgKey: 'dashboard.card.deployable.completed',
      count: 3,
    },
    {
      msgKey: 'dashboard.card.deployable.failed',
      count: 1,
    },
    {
      msgKey: 'dashboard.card.deployable.inProgress',
      count: 1,
    },
    {
      msgKey: 'dashboard.card.deployable.pending',
      count: 0,
    },
  ];
  const countsCardDataIncidents = [
    {
      msgKey: 'dashboard.card.deployable.incidents',
      count: 1,
    },
  ];

  return {
    countsCardDataGeneralInfo,
    countsCardDataIncidents,
  };
};

class ApplicationDeployableHighlights extends React.Component {
  componentDidMount() { }
  componentWillUnmount() { }

  render() {
    const { locale } = this.context;
    const { countsCardDataGeneralInfo, countsCardDataIncidents } = this.props;

    return (
      <React.Fragment>
        <div id="ApplicationDeployableHighlights">
          <div className="deployable-highlights-header">
            {msgs.get('description.title.deployableHighlights', locale)}
          </div>

          <div className="deployable-highlights-container">
            <div className="deployable-highlights-info">
              <CountsCardModule
                data={countsCardDataGeneralInfo}
              />
            </div>
            <div className="deployable-highlights-incidents">
              <CountsCardModule
                data={countsCardDataIncidents}
              />
            </div>
          </div>
        </div>
      </React.Fragment>
    );
  }
}

export default connect(mapStateToProps)(ApplicationDeployableHighlights);
