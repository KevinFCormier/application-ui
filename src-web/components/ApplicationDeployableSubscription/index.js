/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/

import React from '../../../node_modules/react';
import msgs from '../../../nls/platform.properties';
import { withLocale } from '../../providers/LocaleProvider';
import resources from '../../../lib/shared/resources';
import { Icon } from 'carbon-components-react';

resources(() => {
  require('./style.scss')
})

const ApplicationDeployableSubscription = withLocale(({ subscription, locale }) => {
  return (
    <div id="ApplicationDeployableSubscription">
      <div className="deployable-subscription-header">
        {msgs.get('description.title.deployableSubscription', locale)}
      </div>

      <div className="deployable-subscription-container">
        <div className="deployable-subscription-edit-container">
          <p className="yamlEditIconTitle">YAML</p>
          <Icon
            name="icon--edit"
            fill="#6089bf"
            description=""
            className="yamlEditIcon"
          />
        </div>
        <div className="deployable-subscription-tile-container">
          <div className="subscription-name">
            <p className="tile-title">{msgs.get('description.title.deployableSubscription.subscriptionName')}</p>
            <span className="tile-content">{subscription.name ? subscription.name : "--"}</span>
          </div>
          <div className="landing-channel">
            <p className="tile-title">{msgs.get('description.title.deployableSubscription.landingChannel')}</p>
            <span className="tile-content">{subscription.channel ? subscription.channel : "--"}</span>
          </div>
          <div className="certificates">
            <p className="tile-title">{msgs.get('description.title.deployableSubscription.certificates')}</p>
            <span className="tile-content">
              {subscription.certificates ? subscription.certificates + " (" : "--"}
              <a href="#">{subscription.certificates ? msgs.get('routes.viewDetails') : ""}</a>
              {subscription.certificates ? ")" : ""}
            </span>
          </div>
          <div className="annotations">
            <p className="tile-title">{msgs.get('description.title.deployableSubscription.annotations')}</p>
            <span className="tile-content">{subscription.annotations ? subscription.annotations : "--"}</span>
          </div>
          <div className="placement-rules">
            <p className="tile-title">{msgs.get('description.title.deployableSubscription.placementRules')}</p>
            <span className="tile-content">{subscription.rules ? subscription.rules : "--"}</span>
          </div>
          <div className="version">
            <p className="tile-title">{msgs.get('description.title.deployableSubscription.version')}</p>
            <span className="tile-content">{subscription.version ? subscription.version : "--"}</span>
          </div>
          <div className="labels">
            <p className="tile-title">{msgs.get('description.title.deployableSubscription.labels')}</p>
            <div className="tile-content">
              {subscription.labels ? subscription.labels.map(item => (
                <div className="label-item" key={item[0]}>{item[0]}={item[1]}</div>
              ))
                : "--"}
            </div>
          </div>
          <div className="rolling-updates">
            <p className="tile-title">{msgs.get('description.title.deployableSubscription.rollingUpdates')}</p>
            <span className="tile-content">{subscription.updates ? subscription.updates : "--"}</span>
          </div>
        </div>
      </div>
    </div>
  )
})

export default withLocale(ApplicationDeployableSubscription)
