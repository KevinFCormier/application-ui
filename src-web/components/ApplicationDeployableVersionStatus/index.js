/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2017, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/

import React from '../../../node_modules/react';
import msgs from '../../../nls/platform.properties';
import { withLocale } from '../../providers/LocaleProvider';
import { connect } from '../../../node_modules/react-redux';
import resources from '../../../lib/shared/resources';

resources(() => {
  require('./style.scss');
});

const mapStateToProps = (state) => {
  const { } = state;
  return {};
};


const ApplicationDeployableVersionStatus = withLocale(({ locale }) => {

  return (<div id="ApplicationDeployableVersionStatus">
    <div className="deployable-versionStatus-header">
      {msgs.get('description.title.deployableVersionStatus', locale)}
    </div>
  </div>);

});

export default withLocale(ApplicationDeployableVersionStatus);
