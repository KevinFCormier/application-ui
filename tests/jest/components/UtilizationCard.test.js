/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
'use strict'

import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'

import { UtilizationCard } from '../../../src-web/components/UtilizationCard'

describe('UtilizationCard component 1', () => {
  it('renders as expected', () => {
    const component = renderer.create(
      <MemoryRouter initialEntries={['/random']}>
        <UtilizationCard
          critical={1}
          healthy={1}
          warning={1}
          title="unit test"
          type="pods"
        />
      </MemoryRouter>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})

describe('UtilizationCard component 2', () => {
  it('renders as expected', () => {
    const component = renderer.create(
      <MemoryRouter initialEntries={['/random']}>
        <UtilizationCard
          critical={0}
          healthy={0}
          warning={1}
          title="unit test 2"
          type="pods"
        />
      </MemoryRouter>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})

describe('UtilizationCard component 3', () => {
  it('renders as expected', () => {
    const component = renderer.create(
      <MemoryRouter initialEntries={['/random']}>
        <UtilizationCard
          critical={0}
          healthy={1}
          warning={0}
          title="unit test 3"
          type="pods"
        />
      </MemoryRouter>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})

describe('UtilizationCard component 4', () => {
  it('renders as expected', () => {
    const component = renderer.create(
      <MemoryRouter initialEntries={['/random']}>
        <UtilizationCard
          critical={0}
          healthy={0}
          warning={0}
          title="unit test 4"
          type="pods"
        />
      </MemoryRouter>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
