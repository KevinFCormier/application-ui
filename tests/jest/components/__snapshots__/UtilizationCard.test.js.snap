// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UtilizationCard component 1 renders as expected 1`] = `
<div
  className="bx--module bx--module--single bx--tile dashboard-card"
  critical={1}
  healthy={1}
  title="unit test"
  type="pods"
  warning={1}
>
  <div
    className="bx--module__inner"
  >
    <div
      className="bx--module__header dashboard-card-header__utilization"
    >
      <h1
        className="bx--module__title"
      >
        unit test
      </h1>
    </div>
    <div
      className="bx--module__content dashboard-card-orb"
    >
      <div
        className="dashboard-overview"
      >
        <div>
          <div
            className="dashboard-count "
          >
            <div
              className="dashboard-orb"
            >
              <div
                className="dashboard-orb dashboard-orb__inner dashboard-orb__util-critical"
              >
                1
              </div>
            </div>
            <p
              className="dashboard-card-text"
            >
              Above 90%
            </p>
          </div>
          <div
            className="dashboard-count "
          >
            <div
              className="dashboard-orb"
            >
              <div
                className="dashboard-orb dashboard-orb__inner dashboard-orb__util-warning"
              >
                1
              </div>
            </div>
            <p
              className="dashboard-card-text"
            >
              75-90%
            </p>
          </div>
          <div
            className="dashboard-count "
          >
            <div
              className="dashboard-orb"
            >
              <div
                className="dashboard-orb dashboard-orb__inner dashboard-orb__util-healthy"
              >
                1
              </div>
            </div>
            <p
              className="dashboard-card-text"
            >
              Below 75%
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="bx--module__content dashboard-card-table-body"
    >
      <div
        className="dashboard-card-separator-link"
      >
        <a
          href="undefined/clusters"
          onClick={[Function]}
        >
          View All
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`UtilizationCard component 2 renders as expected 1`] = `
<div
  className="bx--module bx--module--single bx--tile dashboard-card"
  critical={0}
  healthy={0}
  title="unit test 2"
  type="pods"
  warning={1}
>
  <div
    className="bx--module__inner"
  >
    <div
      className="bx--module__header dashboard-card-header__utilization"
    >
      <h1
        className="bx--module__title"
      >
        unit test 2
      </h1>
    </div>
    <div
      className="bx--module__content dashboard-card-orb"
    >
      <div
        className="dashboard-overview"
      >
        <div>
          <div
            className="dashboard-count "
          >
            <div
              className="dashboard-orb"
            >
              <div
                className="dashboard-orb dashboard-orb__inner dashboard-orb__gray"
              >
                0
              </div>
            </div>
            <p
              className="dashboard-card-text dashboard-card-text__gray"
            >
              Above 90%
            </p>
          </div>
          <div
            className="dashboard-count "
          >
            <div
              className="dashboard-orb"
            >
              <div
                className="dashboard-orb dashboard-orb__inner dashboard-orb__util-warning"
              >
                1
              </div>
            </div>
            <p
              className="dashboard-card-text"
            >
              75-90%
            </p>
          </div>
          <div
            className="dashboard-count "
          >
            <div
              className="dashboard-orb"
            >
              <div
                className="dashboard-orb dashboard-orb__inner dashboard-orb__gray"
              >
                0
              </div>
            </div>
            <p
              className="dashboard-card-text dashboard-card-text__gray"
            >
              Below 75%
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="bx--module__content dashboard-card-table-body"
    >
      <div
        className="dashboard-card-separator-link"
      >
        <a
          href="undefined/clusters"
          onClick={[Function]}
        >
          View All
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`UtilizationCard component 3 renders as expected 1`] = `
<div
  className="bx--module bx--module--single bx--tile dashboard-card"
  critical={0}
  healthy={1}
  title="unit test 3"
  type="pods"
  warning={0}
>
  <div
    className="bx--module__inner"
  >
    <div
      className="bx--module__header dashboard-card-header__utilization"
    >
      <h1
        className="bx--module__title"
      >
        unit test 3
      </h1>
    </div>
    <div
      className="bx--module__content dashboard-card-orb"
    >
      <div
        className="dashboard-overview"
      >
        <div>
          <div
            className="dashboard-count "
          >
            <div
              className="dashboard-orb"
            >
              <div
                className="dashboard-orb dashboard-orb__inner dashboard-orb__gray"
              >
                0
              </div>
            </div>
            <p
              className="dashboard-card-text dashboard-card-text__gray"
            >
              Above 90%
            </p>
          </div>
          <div
            className="dashboard-count "
          >
            <div
              className="dashboard-orb"
            >
              <div
                className="dashboard-orb dashboard-orb__inner dashboard-orb__gray"
              >
                0
              </div>
            </div>
            <p
              className="dashboard-card-text dashboard-card-text__gray"
            >
              75-90%
            </p>
          </div>
          <div
            className="dashboard-count "
          >
            <div
              className="dashboard-orb"
            >
              <div
                className="dashboard-orb dashboard-orb__inner dashboard-orb__util-healthy"
              >
                1
              </div>
            </div>
            <p
              className="dashboard-card-text"
            >
              Below 75%
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="bx--module__content dashboard-card-table-body"
    >
      <div
        className="dashboard-card-separator-link"
      >
        <a
          href="undefined/clusters"
          onClick={[Function]}
        >
          View All
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`UtilizationCard component 4 renders as expected 1`] = `
<div
  className="bx--module bx--module--single bx--tile dashboard-card"
  critical={0}
  healthy={0}
  title="unit test 4"
  type="pods"
  warning={0}
>
  <div
    className="bx--module__inner"
  >
    <div
      className="bx--module__header dashboard-card-header__utilization"
    >
      <h1
        className="bx--module__title"
      >
        unit test 4
      </h1>
    </div>
    <div
      className="bx--module__content dashboard-card-orb"
    >
      <div
        className="dashboard-overview"
      >
        <div>
          <div
            className="dashboard-count "
          >
            <div
              className="dashboard-orb"
            >
              <div
                className="dashboard-orb dashboard-orb__inner dashboard-orb__gray"
              >
                0
              </div>
            </div>
            <p
              className="dashboard-card-text dashboard-card-text__gray"
            >
              Above 90%
            </p>
          </div>
          <div
            className="dashboard-count "
          >
            <div
              className="dashboard-orb"
            >
              <div
                className="dashboard-orb dashboard-orb__inner dashboard-orb__gray"
              >
                0
              </div>
            </div>
            <p
              className="dashboard-card-text dashboard-card-text__gray"
            >
              75-90%
            </p>
          </div>
          <div
            className="dashboard-count "
          >
            <div
              className="dashboard-orb"
            >
              <div
                className="dashboard-orb dashboard-orb__inner dashboard-orb__gray"
              >
                0
              </div>
            </div>
            <p
              className="dashboard-card-text dashboard-card-text__gray"
            >
              Below 75%
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="bx--module__content dashboard-card-table-body"
    >
      <div
        className="dashboard-card-separator-link"
      >
        <a
          href="undefined/clusters"
          onClick={[Function]}
        >
          View All
        </a>
      </div>
    </div>
  </div>
</div>
`;
